shell/shared/demo.js

```diff
@@ -264,8 +264,9 @@ Router.map(function () {
       var appName = 'missing package';
 
       if (thisPackage) {
-        var actionTitle = thisPackage.manifest.actions[0].title.defaultText;
-        appName = appNameFromActionName(actionTitle);
+        var manifest = thisPackage.manifest;
+        var action = manifest.actions[0];
+        appName = (manifest.appTitle && manifest.appTitle.defaultText) || appNameFromActionName(action.title.defaultText);
```

Comment:
Thanks @kentonv for the merge and the feedback. I'll look into http://www.emacswiki.org/emacs/ColumnMarker to avoid the >100-char lines.